### PR TITLE
feat(drive): expand count-index group-by carrier shapes (G1a/G1b/G8a-c)

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
Extends the count-index group-by surface introduced alongside [#3633](https://github.com/dashpay/platform/pull/3633) with new fixture coverage for edge cases of the carrier-aggregate-on-range (ACOR) and `In`-fanout paths, plus the rs-drive infrastructure those fixtures exercise.

## What was done?
New bench fixtures + matching rs-drive support + book chapter:

- **G1a** — `In` on `byBrand` with one absent value. Proof grows by ~255 B for the absence subproof; verifier omits the absent branch from `Entries`.
- **G1b** — high-fanout `In` on `byBrand` (`|IN| = B = 100`). Same shape as G1, scaled up; reveals every byBrand entry.
- **G8a** — bounded carrier + bounded ACOR (range on both axes) with descending walk. Same carrier shape as G8, different op variants on both range commitments.
- **G8b / G8c** — rejection shapes for the two-range carrier with `group_by = [brand, color]` and `group_by = []`. Carrier is opened only for `GroupByRange + single-field group_by`.

Touched code:
- `packages/rs-drive/src/query/drive_document_count_query/drive_dispatcher.rs` — mode-detection from clauses + per-mode dispatch for the new variants.
- `packages/rs-drive/src/query/drive_document_count_query/{execute_range_count,path_query,tests}.rs` and `executors/range_aggregate_carrier_proof.rs` — executor side.
- `packages/rs-drive/src/query/mod.rs` — query plumbing.
- `packages/rs-drive/src/verify/document_count/verify_carrier_aggregate_count_proof/{mod,v0/mod}.rs` — verifier side.
- `packages/rs-drive/src/drive/contract/insert/insert_contract/v0/mod.rs` — count-tree wiring needed by the new fixtures.
- `packages/rs-drive/benches/document_count_worst_case.rs` — fixture bodies.
- `book/src/drive/count-index-group-by-examples.md` — chapter expanded with the new G* sections (proof sizes, verified shapes, tree-walk descriptions).

## How Has This Been Tested?
- `cargo test -p drive` (executor + verifier paths for the new variants)
- Bench fixture (`document_count_worst_case`) regenerated; proof sizes match the documented G* numbers in the book chapter.

## Breaking Changes
None. Drive-internal additions on top of the v1 count surface; no wire-protocol or consensus changes.

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

## Summary by CodeRabbit

* **New Features**
  * Added support for specifying sort order (ascending/descending) in document count queries with range-based filtering.

* **Improvements**
  * Enhanced range-based document counting operations to respect specified iteration directions.
  * Improved handling of range query constraints in document counting scenarios.

* **Tests**
  * Expanded test suite coverage for document count operations, including additional group-by and range query edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/dashpay/platform/pull/3652?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by thepastaclaw** [comment]
### Review Gate

Commit: `516be5c0`

- ⏳ **Debounce**: 25m ago (need 30m)
- ⏳ **CI checks**: build failure: codecov/patch
- ✅ **CodeRabbit review**: comment found
- ✅ **Off-peak hours**: off-peak (07:51 PM PT Saturday)

- [ ] Run review now (check to override)

File: packages/rs-drive/benches/document_count_worst_case.rs [hunk 1]
```diff
@@ -959,6 +1053,28 @@ fn report_group_by_matrix(fixture: &CountBenchFixture, platform_version: &Platfo
...
+                    "brand",
+                    "<",
+                    Value::Text(brand_label(BRAND_COUNT * 65 / 100)),
+                ),
+                clause("color", ">", Value::Text(color_label(200))),
+                clause("color", "<", Value::Text(color_label(400))),
+            ]),
+            mode: CountMode::GroupByRange,
+            limit: None,
+        },
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive/benches/document_count_worst_case.rs:1107]: Done in 516be5c — added `raw_order_by` to `MatrixCase`, threaded it through `drive_count_outcome`, and set the G8a row to `order_by_brand_desc` so the descending walk actually exercises `left_to_right = false`. Default ascending path preserved for the other 23 cases via `Value::Null`.

**Review by QuantumExplorer — COMMENTED** [review]
Reviewed
